### PR TITLE
Upgrade to 16.04

### PR DESCRIPTION
**Pull request by kiliant**
Another try at upgrading to 16.04, as tests work from clean builds just fine locally.

Forked from #532.

### CODE REVIEW

**Comment by kiliant** [comment]
Weird thing is, that the local checks should have failed, too. 
```
run docker exec mail_override_hostname /bin/sh -c "ls -A /var/mail/localhost.localdomain/user1/new | wc -l | grep 1"
```
fails, but that is not shown by the test run (bug?).
The Makefile has been fixed and the tests have been adapted.

File: Dockerfile [hunk 1]
```diff
@@ -8,10 +8,14 @@ ENV ONE_DIR=0
 # Packages
 RUN apt-get update -q --fix-missing && \
   apt-get -y upgrade && \
+  apt-get -y install postfix
+RUN DEBIAN_FRONTEND=noninteractive apt-get update -q --fix-missing && \
```
Comment (tomav) [inline Dockerfile]: `DEBIAN_FRONTEND=noninteractive` already defined line 4

File: test/config/dovecot-lmtp/local.conf [hunk 2]
```diff
@@ -1 +1,2 @@
 mail_max_userip_connections = 69
+recipient_delimiter = ~
```
Comment (tomav) [inline test/config/dovecot-lmtp/local.conf]: This is not `+` in postfix?

**Comment by tomav** [comment]
No BC Break detected?

**Comment by kiliant** [comment]
These were changes introduced by @42wim.

**Comment by 42wim** [comment]
The dovecot ones are not from me btw. Check my original PR

File: Dockerfile (hunk 1 above)
Comment (42wim) [inline Dockerfile]: Iirc this was the only way it worked. Putting it in the already defined one doesn't complete postfix install

**Comment by kiliant** [comment]
You're right sorry. Weird, I did not change that either. Reversing.

**Comment by kiliant** [comment]
Those changes were caused by running the tests locally and slipped through the commit - now removed.

**Comment by tomav** [comment]
This will be `2.3`.

**Comment by tomav** [comment]
Thanks @kiliant and @42wim

**Pull request merged**